mempool/src/core_mempool/ttl_cache.rs

```diff
@@ -36,16 +36,15 @@ where
     }
 
     pub fn insert(&mut self, key: K, value: V) {
-        // Remove old entry if it exists.
-        match self.data.get(&key) {
+        // Remove the old entry from data and ttl_index (if it exists)
+        match self.data.remove(&key) {
```

Comment:
So this is the only non-refactor of it.  Could we have a test for this?

Reply:
Yeah, this is the only change. Everything else was to make it more obvious that we're. Given that we won't ever hit this in practice unless: (i) the system time is screwed up; or (ii) someone sets an invalid config timeout, I'm not gonna bother. I think it'll be too much work for what it's worth.